ocs_ci/ocs/dr_upgrade.py

```diff
@@ -0,0 +1,271 @@
+            f'"{self.channel}", "source": "{mco_source}"}}}}\''
+        )
+        subscription.exec_oc_cmd(patch_subscription_cmd, out_yaml_format=False)
+
+    def validate_upgrade(self):
+        # In case of both MCO and DRhub operator, validation steps are similar
+        # just the resource names changes
+        assert (
+            self.post_upgrade_data["pod_status"] == "Running"
+        ), f"Pod {self.pod_name_pattern} not in Running state post upgrade"
+        assert (
+            self.post_upgrade_data["age"] <= self.pre_upgrade_data["age"]
+        ), f"{self.pod_name_pattern} didn't restart after upgrade"
+        assert (
+            self.post_upgrade_data["version"] != self.pre_upgrade_data["version"]
+        ), "CSV version not upgraded"
+        check_all_csvs_are_succeeded(namespace=self.namespace)
+
+    def collect_data(self):
+        """
+        Collect DR operator related pods and csv data
+        """
+        pod_data = pod.get_all_pods(namespace=self.namespace)
+        for p in pod_data:
+            if self.pod_name_pattern in p["metadata"]["name"]:
+                pod_obj = OCP(
+                    namespace=self.namespace,
+                    resource_name=p["metadata"]["name"],
+                    kind="Pod",
+                )
+                if self.upgrade_phase == "pre_upgrade":
+                    self.pre_upgrade_data["age"] = pod_obj.get_resource(column="AGE")
+                    self.pre_upgrade_data["pod_status"] = pod_obj.get_resource_status()
+                if self.upgrade_phse == "post_upgrade":
+                    self.post_upgrade_data["age"] = pod_obj.get_resource(column="AGE")
+                    self.post_upgrade_data["pod_status"] = pod_obj.get_resource_status()
+
+        # get pre-upgrade csv for MCO
+        csv_objs = CSV(namespace=self.namespace)
+        for csv in csv_objs:
+            if self.operator_name in csv["metadata"]["name"]:
+                csv_obj = CSV(
+                    namespace=self.namespace, resource_name=csv["metadata"]["name"]
+                )
+                if self.upgrade_phase == "pre_upgrade":
+                    self.pre_upgrade_data["version"] = csv_obj.get_resource(
+                        column="VERSION"
+                    )
+                if self.upgrade_phase == "post_upgrade":
+                    self.post_upgrade_data["version"] = csv_obj.get_resource(
+                        column="VERSION"
+                    )
+        # Make sure all csvs are in succeeded state
+        check_all_csvs_are_succeeded(namespace=self.namespace)
+
+
+class MultiClusterOrchestratorUpgrade(DRUpgrade):
+    """
+    A class to handle ODF MCO operator upgrades
+
+    """
+
+    def __init__(self):
+        super.__init__()
+        self.operator_name = defaults.MCO_OPERATOR_NAME
+        self.subscription_name = constants.MCO_SUBSCRIPTION
+
+    def run_upgrade(self):
```

Comment:
In the case of Z-stream upgrade. We should validate upgrade of DR components in ODF upgrade test case which means.

Before we update catalogSource in ODF upgrade test - we should add code to check if it's Multicluster DR upgrade, if so, init the object of this class, run collect_data and continue with ODF upgrade as usual, once we finish ODF upgrade -> run validation which depends on collected date before upgrade.